src/Equinox/Equinox.fs

```diff
@@ -10,42 +10,50 @@ type MaxResyncsExhaustedException(count) =
 /// Central Application-facing API. Wraps the handling of decision or query flows in a manner that is store agnostic
 type Stream<'event, 'state>
     (   log, stream : IStream<'event, 'state>, maxAttempts : int,
-        [<Optional; DefaultParameterValue(null)>] ?createAttemptsExhaustedException,
+        [<Optional; DefaultParameterValue(null)>] ?createAttemptsExhaustedException : int -> exn,
```

Comment:
this signature fix should be a separate PR, and probably should only happen in V3
(the restriction for the exception to inherit from `MaxResyncsExhaustedException` was unintentional)

Reply:
opted to leave it here